src/server/req_delete.c

```diff
@@ -525,56 +521,48 @@ req_deletejob(struct batch_request *preq)
 
 	++preq->rq_refct;
 	while (1) {
-		if ((i = parse_subjob_index(range, &pc, &x, &y, &z, &j)) == -1) {
+		int start;
+		int end;
+		int step;
+		int count;
+
+		if ((i = parse_subjob_index(range, &pc, &start, &end, &step, &count)) == -1) {
 			req_reject(PBSE_IVALREQ, 0, preq);
 			break;
 		} else if (i == 1)
 			break;
-		/* Ensure that the range specified in the delete job request does not exceed the
-		 index of the highest numbered array subjob */
 
-		count = parent->ji_ajtrk->tkm_ct;
-		maxindex = parent->ji_ajtrk->tkm_tbl[count-1].trk_index;
-		if (x > maxindex) {
+		/*
+		 * Ensure that the range specified in the delete job request does not exceed the
+		 * index of the highest numbered array subjob
+		 */
+		if (start > SJ_TBLIDX_2_IDX(parent, parent->ji_ajtrk->tkm_ct - 1)) {
 			req_reject(PBSE_UNKJOBID, 0, preq);
 			break;
 		}
-		while (x <= y) {
-			i = numindex_to_offset(parent, x);
-			if (i < 0) {
-				x += z; /* no such index, ignore it */
-				continue;
-			}
+		for (i = start; i <= end; i += step) {
+			int idx = SJ_IDX_2_TBLIDX(parent, i);
 
-			sjst = get_subjob_state(parent, i);
-			if ((sjst == JOB_STATE_EXITING) && !forcedel) {
-				x += z; /* ignore it */
+			sjst = get_subjob_state(parent, idx);
+			if ((sjst == JOB_STATE_EXITING) && !forcedel)
```

Comment:
You are doing a continue in a while loop without changing anything.  This could lead to an infinite loop.

Reply:
@bhroam I changed while loop to for loop, so I don't think it will be an infinite loop.